### PR TITLE
Feature/sg 70 kd loss

### PR DESCRIPTION
**Pull request by shairoz-deci**
This PR now holds the following changes associated with kd training:\
- Loss
- context update method (inheritance)
- Training unit tests

### CODE REVIEW

File: src/super_gradients/training/sg_model/sg_model.py [hunk 1]
```diff
@@ -1669,6 +1671,16 @@ def evaluate(self, data_loader: torch.utils.data.DataLoader, metrics: MetricColl
...
 
+    @staticmethod
+    def update_context(context: PhaseContext, **kwargs):
+        """
+        Updates phase context with **kwargs
+
+        :param context: phase context to be updated
+        :param kwargs: kwargs to be set as the context's attributes.
+        """
+        context.update_context(**kwargs)
```
Comment (lotem-deci) [inline src/super_gradients/training/sg_model/sg_model.py:1682]: @shaydeci  - This does not feel right... Let's talk about this tomorrow

**Review by lotem-deci — APPROVED** [review]
LGTM
